src/sync/sync_config.hpp

```diff
@@ -100,6 +104,7 @@ struct SyncConfig {
     SyncSessionStopPolicy stop_policy;
     std::function<SyncBindSessionHandler> bind_session_handler;
     std::function<SyncSessionErrorHandler> error_handler;
+    std::shared_ptr<ChangesetTransformer> transformer;
```

Comment:
What is the expected lifetime of the `ChangesetTransformer` with respect to other objects? Having the configuration share ownership while the `sync::Session` does nothing to keep it alive seems a bit weird.